File: app/src/main/java/org/mifos/mobile/ui/views/LoanApplicationMvpView.java
package org.mifos.mobile.ui.views;

import org.mifos.mobile.models.templates.loans.LoanTemplate;
import org.mifos.mobile.ui.views.base.MVPView;

/**
 * Created by Rajan Maurya on 06/03/17.
 */

public interface LoanApplicationMvpView extends MVPView {

    void showUserInterface();

    void showLoanTemplate(LoanTemplate loanTemplate);

    void showUpdateLoanTemplate(LoanTemplate loanTemplate);

    void showLoanTemplateByProduct(LoanTemplate loanTemplate);

    void showUpdateLoanTemplateByProduct(LoanTemplate loanTemplate);

    void showError(String message);
}

File: app/src/main/java/org/mifos/mobile/ui/views/LoanApplicationMvpView.kt
package org.mifos.mobile.ui.views

import org.mifos.mobile.models.templates.loans.LoanTemplate
import org.mifos.mobile.ui.views.base.MVPView

/**
 * Created by Rajan Maurya on 06/03/17.
 */
interface LoanApplicationMvpView : MVPView {
    fun showUserInterface()
    fun showLoanTemplate(loanTemplate: LoanTemplate?)
    fun showUpdateLoanTemplate(loanTemplate: LoanTemplate?)
    fun showLoanTemplateByProduct(loanTemplate: LoanTemplate?)
    fun showUpdateLoanTemplateByProduct(loanTemplate: LoanTemplate?)
    fun showError(message: String?)
}